src/lib.rs

```diff
@@ -240,105 +329,255 @@ pub enum Delimiter {
     None,
 }
 
-#[derive(Copy, Clone)]
-pub struct Term(imp::Term);
+impl Group {
+    pub fn new(delimiter: Delimiter, stream: TokenStream) -> Group {
+        Group { delimiter, stream, span: Span::call_site() }
+    }
 
-impl Term {
-    pub fn intern(string: &str) -> Term {
-        Term(imp::Term::intern(string))
+    pub fn delimiter(&self) -> Delimiter {
+        self.delimiter
     }
 
-    pub fn as_str(&self) -> &str {
-        self.0.as_str()
+    pub fn stream(&self) -> &TokenStream {
```

Comment:
This is sort of an unfortunate signature because you can't do anything with &TokenStream except clone or check if empty. Would it work to implement IntoIterator for &TokenStream? Or is there a cheap way to return TokenStream here?